docs/password_hashing.rst

```diff
@@ -0,0 +1,136 @@
+.. _password-hashing:
+
+Password hashing
+================
+
+.. currentmodule:: nacl.pw_hash
+
+An ever important use of cryptographic hashing primitives has been
+the password hashing one, starting from the early 1970's years, at
+first just to avoid storing clear-text passwords.
+
+To make a long story short [SD2012]_, password hashing and password based key
+derivation mechanisms in actual use are all based on the idea of iterating
+many times a hash function on a combination of the password and
+a random ``salt`` which is stored along with the hash, and allows
+verifying a proposed password while avoiding clear-text storage.
+
+The latest developments in password hashing have been mechanisms
```

Comment:
I'd prefer giving a more neutral description of scrypt, since I'm thinking to add argon2i bindings too (not as 1.1 material in my opinion)